spring-social-zotero/src/main/java/org/springframework/social/zotero/api/impl/GroupsTemplate.java

```diff
@@ -350,4 +413,34 @@ public void deleteItem(String groupId, String citationKey, Long citationVersion)
         }
 
     }
+    
+    private ZoteroUpdateItemsStatuses getStatusesFromResponse(List<ItemCreationResponse> responses,
+            List<String> itemsKeys) {
+        // Zotero is not sending failed items keys in response. So, we have to compute
+        // failed items keys
+        // on our own using success and unchanged keys
+
+        ZoteroUpdateItemsStatuses statuses = new ZoteroUpdateItemsStatuses();
+        List<String> successKeys = new ArrayList<>();
+        List<String> failedKeys = new ArrayList<>();
+        List<String> unchangedKeys = new ArrayList<>();
+        for (ItemCreationResponse response : responses) {
+            Function<Map.Entry<String, String>, String> itemKeyExtractor = e -> e.getValue();
```

Comment:
I would put the lambda directly in the `extractItemKeys` methods even if it is technically duplicate code. It's easier to read/understand than this line.